homeassistant/components/apple_tv/config_flow.py

```diff
@@ -0,0 +1,442 @@
+        creds = {
+            CREDENTIAL_MAPPING[prot]: creds
+            for prot, creds in info.get(CONF_CREDENTIALS).items()
+        }
+        return await self._async_get_entry(
+            info.get(CONF_IDENTIFIER),
+            const.Protocol[info.get(CONF_PROTOCOL)],
+            info.get(CONF_NAME),
+            creds,
+            info.get(CONF_ADDRESS),
+            is_import=True,
+        )
+
+    async def _async_get_entry(
+        self, identifier, protocol, name, credentials, address, is_import=False
+    ):
+        if not is_valid_credentials(credentials):
+            return self.async_abort(reason="invalid_config")
+
+        data = {
+            CONF_IDENTIFIER: identifier,
+            CONF_PROTOCOL: protocol.value,
+            CONF_NAME: name,
+            CONF_CREDENTIALS: credentials,
+            CONF_ADDRESS: str(address),
+        }
+
+        config_entry = self._get_config_entry(identifier)
+        if config_entry:
+            config_entry.data.update(data)
+            self.hass.config_entries.async_update_entry(config_entry)
+            return self.async_abort(reason="updated_configuration")
+
+        title = name + (" (import from configuration.yaml)" if is_import else "")
+        return self.async_create_entry(title=title, data=data)
+
+    def _next_protocol_to_pair(self):
+        def _needs_pairing(protocol):
+            if self.atv.get_service(protocol) is None:
+                return False
+            return protocol.value not in self.credentials
+
+        for protocol in PROTOCOL_PRIORITY:
+            if _needs_pairing(protocol):
+                return protocol
+        return None
+
+    def _devices_str(self):
+        return ", ".join(
+            [
+                f"`{x.name} ({x.address})`"
+                for x in self.scan_result
+                if not self._is_already_configured(x.identifier)
+            ]
+        )
+
+    def _prefill_identifier(self):
+        # Return identifier (address) of one device that has not been paired with
+        for atv in self.scan_result:
+            if not self._is_already_configured(atv.identifier):
+                return str(atv.address)
+        return ""
+
+    def _is_already_configured(self, identifier):
+        return self._get_config_entry(identifier) is not None
+
+    def _get_config_entry(self, identifier):
+        for entry in self._async_current_entries():
+            if entry.data[CONF_IDENTIFIER] == identifier:
+                return entry
+        return None
+
+
+class AppleTVOptionsFlow(config_entries.OptionsFlow):
+    """Handle Apple TV options."""
+
+    def __init__(self, config_entry):
+        """Initialize Apple TV options flow."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+
+    async def async_step_init(self, user_input=None):
+        """Manage the Apple TV options."""
+        return await self.async_step_device_options()
```

Comment:
This is not actually a requirement, this is more often done to future proof the options and translations if additional types of options where to be added at a later point in time

Reply:
Yeah, it seems a bit excessive now that I look at it.